utils/db.ts

```diff
@@ -144,80 +144,74 @@ export function listItemsByTime(options?: Deno.KvListOptions) {
 
 // Notification
 export interface Notification {
+  // Uses ULID
+  id: string;
   userLogin: string;
   type: string;
   text: string;
   originUrl: string;
-  // The below properties can be automatically generated upon notification creation
-  id: string;
-  createdAt: Date;
-}
-
-export function newNotificationProps(): Pick<Item, "id" | "createdAt"> {
-  return {
-    id: crypto.randomUUID(),
-    createdAt: new Date(),
-  };
 }
 
 /**
  * Creates a new notification in KV. Throws if the item already exists in one of the indexes.
  *
  * @example
  * ```ts
- * import { newNotificationProps, createNotification } from "@/utils/db.ts";
+ * import { createNotification } from "@/utils/db.ts";
+ * import { monotonicUlid } from "std/ulid/mod.ts";
  *
  * await createNotification({
+ *   id: monotonicUlid(),
  *   userLogin: "john_doe",
  *   type: "example-type",
  *   text: "Hello, world!",
  *   originUrl: "https://hunt.deno.land",
- *   ...newNotificationProps(),
  * });
  * ```
  */
 export async function createNotification(notification: Notification) {
-  const notificationsKey = ["notifications", notification.id];
-  const notificationsByUserKey = [
+  const key = [
     "notifications_by_user",
     notification.userLogin,
-    notification.createdAt.getTime(),
     notification.id,
   ];
 
   const res = await kv.atomic()
-    .check({ key: notificationsKey, versionstamp: null })
-    .check({ key: notificationsByUserKey, versionstamp: null })
-    .set(notificationsKey, notification)
-    .set(notificationsByUserKey, notification)
+    .check({ key: key, versionstamp: null })
+    .set(key, notification)
     .commit();
 
   if (!res.ok) {
     throw new Error(`Failed to create notification: ${notification}`);
   }
 }
 
-export async function deleteNotification(notification: Notification) {
-  const notificationsKey = ["notifications", notification.id];
-  const notificationsByUserKey = [
+export async function deleteNotification(
+  notification: Pick<Notification, "id" | "userLogin">,
+) {
+  const key = [
     "notifications_by_user",
     notification.userLogin,
-    notification.createdAt.getTime(),
     notification.id,
   ];
 
   const res = await kv.atomic()
-    .delete(notificationsKey)
-    .delete(notificationsByUserKey)
+    .delete(key)
```

Comment:
Because `.delete` never throws even when the key doesn't exist, I wonder if it's better to `.check` the key as well.

(The bug of vote a while ago was caused by missing check of deleting key of votes)

Reply:
Ah, good point. I've added the check and tweaked tests.